app/utility/api/serializer.rb

```diff
@@ -0,0 +1,45 @@
+module Api
+  class Serializer
+    attr_reader :attributes
+    MODEL_INFO = YAML.load(File.new(Rails.root.join('config', 'api.yml')).read).freeze
+
+    def initialize(model, exclude: [])
+      @model = model
+      @exclude = Array.wrap(exclude).map(&:to_s)
+      @class_name = @model.class.name.downcase
+      @fields = MODEL_INFO.dig(@class_name, 'fields')&.reject { |f| @exclude.include?(f) }
```

Comment:
`@exclude.include?` is funny. :P

Reply:
on a serious note, i am curious: what is the difference between `exclude`ing a field and `ignore`-ing it?

Reply:
good question.  while they do mostly the same thing, however there's reason for them being those of as distinct: 

fields that are 'ignored' in the configuration are for fields we want to permanently exclude from attributes. Fields such as 'last_user_id' and 'delta' which should never show up in an api call.

The ```exclude``` option allows you to optionally skip additional columns, that for whatever reason you might not want to include in a particular response, although they are still valid columns.

The use case for this so far ``` extensions ``` for entity. The ```extensions``` attribute can sometimes double the size of the API response (and a executes  a handful of extra sql queries). The entity api skips that attribute by default, but it can be requested via an optional parameter, if the user wants extension data.